MobileWallet/UIElements/EmoticonView.swift

```diff
@@ -240,12 +270,50 @@ class EmoticonView: UIView {
         //If they're typing somewhere, close the keyboard
         superVc.view.endEditing(true)
 
-        UIView.animate(withDuration: 0.5, animations: { [weak self] in
+        if shouldShowBlurContainerViewWhenExpanded {
+            UIView.animate(withDuration: 0.5, animations: { [weak self] in
+                guard let self = self else { return }
+                self.containerView.alpha = 0.7
+                self.superVc.navigationController?.navigationBar.layer.zPosition = -1
+                self.layoutIfNeeded()
+            })
+        }
+
+        tapToExpand?()
+
+        scrollView.contentOffset = CGPoint(x: 30, y: 0)
+
+        UIView.animate(withDuration: 0.3, animations: { [weak self] in
             guard let self = self else { return }
-            self.containerView.alpha = 0.7
-            self.superVc.navigationController?.navigationBar.layer.zPosition = -1
+            self.scrollView.contentOffset = CGPoint(x: -30, y: 0)
             self.layoutIfNeeded()
-        })
+        }) { [weak self] (_) in
+            guard let self = self else { return }
+            UIView.animate(withDuration: 0.3, animations: { [weak self] in
+                guard let self = self else { return }
+                self.scrollView.contentOffset = CGPoint(x: 30, y: 0)
+                self.layoutIfNeeded()
+            }) { [weak self] (_) in
+                guard let self = self else { return }
+                UIView.animate(withDuration: 0.3, animations: { [weak self] in
+                    guard let self = self else { return }
+                    self.scrollView.contentOffset = CGPoint(x: -10, y: 0)
+                    self.layoutIfNeeded()
+                }) { [weak self] (_) in
+                    guard let self = self else { return }
+                    self.scrollView.setContentOffset(CGPoint(x: 0, y: 0), animated: true)
+                }
+            }
+        }
+    }
+
+    private func determineEmojiLabelText(emojiText: String) {
+        let firstThreeChar = emojiText.prefix(3)
+        let lastThreeChar = emojiText.suffix(3)
+
+        let expandedString = self.emojiText.insertSeparator(" | ", atEvery: 3)
```

Comment:
If this color is used in a shared component it should be named something more related to this component. As it's not the create wallet specific color.